plugins/basestate/BaseState.cc

```diff
@@ -0,0 +1,243 @@
+        if (_ecm.HasEntity(baseLinkEntity))
+        {
+            m_baseLinkEntity = baseLinkEntity;
+            yInfo() << "gz-sim-yarp-basestate-system: base link name '" << baseLinkName
+                    << "' found.";
+        } else
+        {
+            yError() << "gz-sim-yarp-basestate-system: base link name '" << baseLinkName
+                     << "' configured was not found in the model definition.";
+            return;
+        }
+
+        // Create link from entity
+        m_baseLink = Link(m_baseLinkEntity);
+        // Enable velocity computation in Gazebo
+        m_baseLink.EnableVelocityChecks(_ecm, true);
+        // Enable acceleration computation in Gazebo
+        m_baseLink.EnableAccelerationChecks(_ecm, true);
+
+        m_baseLinkScopedName = scopedName(m_baseLinkEntity, _ecm);
+        m_baseStateData.baseLinkScopedName = m_baseLinkScopedName;
+
+        driver_properties.put(YarpBaseStateScopedName.c_str(), m_baseLinkScopedName.c_str());
+
+        if (!driver_properties.check("yarpDeviceName"))
+        {
+            yError() << "gz-sim-yarp-basestate-system : missing yarpDeviceName parameter for "
+                        "device "
+                     << m_baseLinkScopedName;
+            return;
+        }
+
+        // Insert the pointer in the singleton handler for retrieving it in the yarp driver
+        BaseStateDataSingleton::getBaseStateDataHandler()->setBaseStateData(&(m_baseStateData));
+
+        driver_properties.put("device", "gazebo_basestate");
+        driver_properties.put("robot", m_baseLinkScopedName);
+
+        if (!m_baseStateDriver.open(driver_properties))
+        {
+            yError() << "gz-sim-yarp-basestate-system Plugin failed: error in opening yarp "
+                        "driver";
+            return;
+        }
+
+        m_deviceScopedName
+            = m_baseLinkScopedName + "/" + driver_properties.find("yarpDeviceName").asString();
+
+        if (!Handler::getHandler()->setDevice(m_deviceScopedName, &m_baseStateDriver))
+        {
+            yError() << "gz-sim-yarp-basestate-system: failed setting scopedDeviceName(="
+                     << m_deviceScopedName << ")";
+            return;
+        }
+        m_deviceRegistered = true;
+        yInfo() << "Registered YARP device with instance name:" << m_deviceScopedName;
+    }
+
+    virtual void PostUpdate(const UpdateInfo& _info, const EntityComponentManager& _ecm) override
+    {
+        bool dataAvailable = true;
+
+        math::Pose3d worldBasePose;
+        math::Vector3d worldBaseLinVel;
+        math::Vector3d worldBaseAngVel;
+        math::Vector3d worldBaseLinAcc;
+        math::Vector3d worldBaseAngAcc;
+
+        // Get the pose of the origin of the link frame in the world reference frame
+        if (dataAvailable && m_baseLink.WorldPose(_ecm).has_value())
+            worldBasePose = m_baseLink.WorldPose(_ecm).value();
+        else
+            dataAvailable = false;
```

Comment:
This is just something that is matter of taste, so do as you prefer. But I typically find if/else bodies without any bracket to be quite confusing, see https://stackoverflow.com/a/2125078 for the reason why.

Reply:
Sure, I used to do it, but I agree that is less future-proof (btw interesting article!) I will update it.

Reply:
Solved in cc3bcc7ad30d2c1bd32c97c021dc388188fdde17